src/AppStore.ts

```diff
@@ -50,14 +33,23 @@ export class AppStore {
         }
     }
 
-    @computed get undismissedSiteErrors() {
+    @computed get undismissedSiteErrors(): SiteError[] {
         return _.filter(this.siteErrors.slice(), err => !err.dismissed);
     }
 
     @computed get isErrorCondition() {
+        // does it have at least one error of type screen (total failure as opposed to recoverable
+        // which we will represent as dialog
         return this.undismissedSiteErrors.length > 0;
     }
 
+    @computed get dialogErrors() {
+        return _.filter(
```

Comment:
small thing but `this.undismissedSiteErrors.filter(...)` is more direct to read, IMO

Reply:
actualy this code is defunct. removing